### PR TITLE
Appnexus geo targeting changes

### PR DESCRIPTION
**Pull request by rakeshbalakrishnan28**

### CODE REVIEW

File: src/app/controllers/appnexus/geoController.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+
+class GeoController{
+
+	async launchGeo(profile) {
+		try {
+			logger.info('********* Launching Geo Targeting *********');
+			let geoResponse = {};
+			const geoTarget = await appnexusModel.getGeoTargets(profile.line_item_id);
+			console.log(geoTarget[0].country_targets);
```
Comment (emx-jesse) [inline src/app/controllers/appnexus/geoController.js]: Can you do the index lookup on line 13? That way you don't need to access index 0 for every geoTarget variable.
Is there ever a scenario where geoTarget will return null/undefined? Is it safe to do these lookups (index 0 + country_targets) without input validation?

File: src/app/controllers/appnexus/lineItemController.js [hunk 2]
```diff
@@ -45,6 +46,11 @@ class LineItemController{
 				return  response = { error: true ,message: 'Budget Intervals not found' }
 			}
 
+            if(lineElement.profile_id){
+                const targeting_profile = await appnexusModel.getId(lineElement.profile_id,'profiles','id');
+                profile_id = targeting_profile[0].appnexus_profile_id;
```
Comment (emx-jesse) [inline src/app/controllers/appnexus/lineItemController.js]: Is there a risk of profile_id being deleted (from APN) if there's some sort of database failure?

File: src/app/controllers/appnexus/lineItemController.js [hunk 3]
```diff
@@ -113,6 +121,10 @@ class LineItemController{
...
```
Comment (emx-jesse) [inline src/app/controllers/appnexus/lineItemController.js]: Same concern as in createLineItem. Is it possible that we'd update the profile_id to null?

File: src/app/controllers/appnexus/geoController.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
+const apnxApi = require('./../../../services/appnexus');
+const appnexusModel = require('./../../models/appnexus');
+const logger = require('../../../commons/logger');
+const dateFormat = require ('dateformat');
```
[the remark this answers is not in the document]
Comment (rakeshbalakrishnan28) [reply src/app/controllers/appnexus/geoController.js]: Done, removed unsued.

File: src/app/controllers/appnexus/geoController.js (hunk 1 above)
Comment (rakeshbalakrishnan28) [reply src/app/controllers/appnexus/geoController.js]: Modified this appropriately.

File: src/app/controllers/appnexus/lineItemController.js (hunk 2 above)
Comment (rakeshbalakrishnan28) [reply src/app/controllers/appnexus/lineItemController.js]: I don't think there should be profile ID deletions on APNX, but now I added some logic to cover bases, where we wont be creating line items if profile_id is not found. Also we might have to cover these cases in our DB

File: src/app/controllers/appnexus/lineItemController.js (hunk 3 above)
Comment (rakeshbalakrishnan28) [reply src/app/controllers/appnexus/lineItemController.js]: Agreed, profile_id cant be null on line item, changed this appropriately, we need to make the profile_id column as a required in table.

File: src/app/controllers/appnexus/lineItemController.js (hunk 2 above)
Comment (emx-jesse) [reply src/app/controllers/appnexus/lineItemController.js]: Sounds good and yeah I agree these should be covered in the DB

Comment (rakeshbalakrishnan28) [reply src/app/controllers/appnexus/lineItemController.js]: Also missed out the one the document/comment, will definitely start adding them asap :D

**Pull request closed without merging**